tools/update_latest_report.py

```diff
@@ -41,63 +43,69 @@
     dir_last_report_rel = sys.argv[2]
 
 dir_report = os.path.join(root, dir_report_rel)
-dir_last_report_symbolic = os.path.join(root, dir_last_report_rel)
-
-report_path = os.path.join(dir_report, "test-status.json")
-
-dir_badge = os.path.join("data/badges", dir_last_report_rel)
-os.makedirs(dir_badge, exist_ok=True)
-
-dir_redirect = os.path.join("gh-pages", dir_last_report_rel)
-os.makedirs(dir_redirect, exist_ok=True)
 
 ################################################################################
 # Read current test-status
+report_path = os.path.join(dir_report, "test-status.json")
 with open(report_path, "r") as f:
     report = json.load(f)
 
-repo = report["repo"]
-
 ################################################################################
 # Update symlink
+dir_last_report_symbolic = os.path.join(root, dir_last_report_rel)
 symlink(dir_report_rel, dir_last_report_symbolic, overwrite=True)
 
-############################################################################
+################################################################################
 # Generate html redirect
-with open(os.path.join(dir_redirect, "redirect.html"), "w") as f:
-    f.write(
+def generate_redirect(report: Dict[str, Any], dir_report: str, dir_last_report_rel: str) -> None:
+    dir_redirect = os.path.join("gh-pages", dir_last_report_rel)
+    os.makedirs(dir_redirect, exist_ok=True)
+
+    repo = report["repo"]
+    blob_url = repo + "/blob/" + dir_report + "/report.md"
+    url = repo + "/" + dir_report + "/report.md"
+
+    with open(os.path.join(dir_redirect, "redirect.html"), "w") as f:
+        f.write(
+            f"""
+        <!DOCTYPE html>
+        <meta charset="utf-8">
+        <title>Redirecting to latest report</title>
+        <meta http-equiv="refresh" content="0; URL={blob_url}">
+        <link rel="canonical" href="{url}">
+        </meta>
+        <body>
+        <a href="{url}">{url}</a>
```

Comment:
Shouldn't this be the following?
```suggestion
        <a href="{blob_url}">{url}</a>
```

Reply:
I don't think so... Well, to be frnak I don't understand what's going on here, and what the difference between `url` and `blob_url` is supposed to be. But I can say that it is damn weird to show a user an URL, but have the URL link to a *different* URL!

But perhaps *both* URLs should be shown to the user?

Alas, ooking at the actual `latest-master/redirect.html` file in the `gh-pages` branch, it seems that `url` is actual not a valid URL (leads to a 404). So again, I don't really understand what's going on here... Perhaps you could clarify?

Reply:
I actually would replace every url with blob_url.